src/pages/index.js

```diff
@@ -51,41 +66,140 @@ enableValidation(options);
 //Functions to pass to class objects
 function submitProfile (evt) {
   evt.preventDefault();
-  user.setUserInfo(profileFormPopup.getInputValues());
-  profileFormPopup.close();
+  buttons.profileSubmitButton.textContent = 'Saving...'
+
+  //TODO: look into finding a way to do this with one call to .getInputValues()
+  const param = {
+    name: profileFormPopup.getInputValues().username,
+    about: profileFormPopup.getInputValues().description
+  };
+  api.patchUserInfo(param)
+    .then((response) => {
+      user.setUserInfo(response)
+    })
+    .catch((response) => {
+      api.catchErrors(response);
+    })
+    .finally(() => {
+      profileFormPopup.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.